File: EventTime/src/main/java/com/hakankuru/EventTime/dto/UserProfileResponse.java
package com.hakankuru.EventTime.dto;

import com.hakankuru.EventTime.entity.GlobalRole;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileResponse {
    private Long userId;
    private String name;
    private String email;
    private GlobalRole globalRole;
    private List<UserClubDTO> clubs;
}

File: eventhub/app/src/main/java/com/hakankuru/eventhub/data/remote/response/UserProfileResponse.kt
package com.hakankuru.eventhub.data.remote.response

data class UserProfileResponse(
    val userId: Long,
    val name: String,
    val email: String,
    val globalRole: String?,
    val clubs: List<UserClubDTO>,
    /** ADMIN için hangi üniversiteyi yönettiği. SUPER_ADMIN ve USER için null. */
    val universityId: Long? = null
)

data class UserClubDTO(
    val clubId: Long,
    val clubName: String,
    val clubRole: String
)
